src/keypair.rs

```diff
@@ -95,40 +92,24 @@ impl Keypair {
     /// # Example
     ///
     /// ```
-    /// extern crate rand;
     /// extern crate ed25519_dalek;
     ///
     /// # #[cfg(feature = "std")]
     /// # fn main() {
     ///
-    /// use rand::rngs::OsRng;
     /// use ed25519_dalek::Keypair;
     /// use ed25519_dalek::Signature;
     ///
-    /// let mut csprng = OsRng{};
-    /// let keypair: Keypair = Keypair::generate(&mut csprng);
+    /// let keypair: Keypair = Keypair::generate();
     ///
     /// # }
     /// #
     /// # #[cfg(not(feature = "std"))]
     /// # fn main() { }
     /// ```
-    ///
-    /// # Input
-    ///
-    /// A CSPRNG with a `fill_bytes()` method, e.g. `rand_os::OsRng`.
-    ///
-    /// The caller must also supply a hash function which implements the
-    /// `Digest` and `Default` traits, and which returns 512 bits of output.
-    /// The standard hash function used for most ed25519 libraries is SHA-512,
-    /// which is available with `use sha2::Sha512` as in the example above.
-    /// Other suitable hash functions include Keccak-512 and Blake2b-512.
     #[cfg(feature = "rand")]
-    pub fn generate<R>(csprng: &mut R) -> Keypair
-    where
-        R: CryptoRng + RngCore,
```

Comment:
Note that this precludes use by embedded users who don't have an `OsRng` and instead have an owned RNG seeded by a hardware peripheral.

Ideally `rand` (or at least `rand_core`) will be stabilized and integrated into the language proper. IMO it would be good to check the `rust-rand` roadmap to see if that's what's planned for the next release. If there's only one more such upgrade, perhaps hiding it from the public API is unnecessary.